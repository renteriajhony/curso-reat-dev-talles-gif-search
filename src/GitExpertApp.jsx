import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GitExpertApp = () => {
  const [categories, setCategorie] = useState(['One Punch Man']);

  const onAddCategory = (category) => {
    categories.find(
      (item) => item.toLowerCase().trim() === category.toLowerCase().trim(),
    )
      ? alert('Category already exists!')
      : setCategorie([category, ...categories]);
  };

  return (
    <>
      <h1>GitExpertApp</h1>
      <AddCategory onAddCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
