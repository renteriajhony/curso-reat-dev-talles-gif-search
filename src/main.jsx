import React from 'react';
import ReactDOM from 'react-dom/client';
import { GitExpertApp } from './GitExpertApp.jsx';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  //TODO: para producción, usar React.StrictMode, que se encarga de detectar errores de diseño, y manejarlos de manera segura
  // <React.StrictMode>
  //   <GitExpertApp />
  // </React.StrictMode>,

  <GitExpertApp />,
);
