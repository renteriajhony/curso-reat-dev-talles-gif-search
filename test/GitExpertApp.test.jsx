import { render, screen } from '@testing-library/react';
// import { fireEvent, render, screen } from '@testing-library/react';
import { GitExpertApp } from '../src/GitExpertApp';

describe('Pruebas en GitExpertApp', () => {
  const initialValue = 100;

  test('debe hacer match con snapshop ', () => {
    const { container } = render(<GitExpertApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test('debe cambiar stado al ejecutar onAddCategory ', () => {
    render(<GitExpertApp value={initialValue} />);
    screen.debug();
  });
});
