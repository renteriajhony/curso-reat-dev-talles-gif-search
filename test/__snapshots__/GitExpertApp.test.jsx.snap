// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GitExpertApp debe cambiar   1`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch Man
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Pruebas en GitExpertApp debe cambiar stado al   1`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch Man
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Pruebas en GitExpertApp debe cambiar stado al ejecutar on  1`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch Man
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Pruebas en GitExpertApp debe cambiar stado al ejecutar onAddCategory  1`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch Man
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Pruebas en GitExpertApp debe hacer match con snapshop  1`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch Man
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Pruebas en GitExpertApp debe hacer match con snapshop  2`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch Man
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;
